src/command/dev/mcp/binary.rs

```diff
@@ -78,6 +77,35 @@ pub struct RunMcpServerBinary<Spawn: Send> {
     env: HashMap<String, String>,
 }
 
+impl<Spawn: Send> RunMcpServerBinary<Spawn> {
+    // Gather the rover-specific configuration options into environment variables
+    // understood by the MCP server.
+    // TODO: Magic strings are not fun to debug later.
+    fn opts_into_env(self) -> HashMap<String, String> {
+        let overlayed = HashMap::from([
+            // Configure the schema to be a local file
+            ("APOLLO_MCP_SCHEMA__SOURCE".to_string(), "local".to_string()),
+            (
+                "APOLLO_MCP_SCHEMA__PATH".to_string(),
+                self.supergraph_schema_path.to_string(),
+            ),
+            // Configure the endpoint from the running router instance
+            (
+                "APOLLO_MCP_ENDPOINT".to_string(),
+                self.router_address.pretty_string(),
+            ),
+            (
+                "APOLLO_MCP_TRANSPORT__TYPE".to_string(),
+                "streamable_http".to_string(),
+            ),
```

Comment:
Sorry but I'm not familiar with these environment variables. How are they used?

Reply:
They come from the new config options for the apollo MCP server. The library we are using for parsing config files allows for overriding options using environment variables using the key `APOLLO_MCP_` followed by the config options, with nesting represented by `__`. 

So in this case, the environment variables set above correspond to the following config:

```yaml
schema:
  source: local
  path: <FROM ROVER>
endpoint: <FROM ROVER>
transport:
  type: streamable_http
```